src/main/kotlin/de/klg71/keycloakmigration/changeControl/actions/role/UpdateRoleAction.kt

```diff
@@ -0,0 +1,88 @@
+package de.klg71.keycloakmigration.changeControl.actions.role
+
+import de.klg71.keycloakmigration.changeControl.actions.Action
+import de.klg71.keycloakmigration.changeControl.actions.MigrationException
+import de.klg71.keycloakmigration.keycloakapi.KeycloakApiException
+import de.klg71.keycloakmigration.keycloakapi.clientRoleByName
+import de.klg71.keycloakmigration.keycloakapi.existsClient
+import de.klg71.keycloakmigration.keycloakapi.model.Role
+import de.klg71.keycloakmigration.keycloakapi.model.RoleListItem
+import de.klg71.keycloakmigration.keycloakapi.roleExistsByName
+
+private const val HTTP_NO_CONTENT = 204
+
+@Suppress("LongParameterList")
+class UpdateRoleAction(
+    realm: String? = null,
+    private val name: String,
+    private val clientId: String? = null,
+    private val description: String = "",
+    private val attributes: Map<String, List<String>>? = null,
+    private val composite: Boolean? = null,
+    private val clientRole: Boolean? = null,
+    private val containerId: String? = null,
+    private val compositeChildRoles: List<RoleSelector>? = null
+) : Action(realm) {
+
+    private lateinit var originalRole: Role
+    private fun updateRole() = Role(
+        originalRole.id,
+        originalRole.name,
+        description,
```

Comment:
can we make description nullable too and only change it if it is set?

Reply:
good catch! adjusted